beacon_node/eth2_libp2p/src/peer_manager/mod.rs

```diff
@@ -521,6 +531,67 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
         }
     }
 
+    pub(crate) fn update_gossipsub_scores(&mut self, gossipsub: &Gossipsub) {
+        //collect peers with scores
+        let mut guard = self.network_globals.peers.write();
+        let mut peers: Vec<_> = guard
+            .peers_mut()
+            .filter_map(|(peer_id, info)| {
+                gossipsub
+                    .peer_score(peer_id)
+                    .map(|score| (peer_id, info, score))
+            })
+            .collect();
+
+        // sort descending by score
+        peers.sort_unstable_by(|(.., s1), (.., s2)| s2.partial_cmp(s1).unwrap_or(Ordering::Equal));
+
+        //check if we are in emergency mode (if a topic has too few non-negative peers)
```

Comment:
So we need to check the logic here. 

A node that has no validator attached to it, will not subscribe to any subnets and will not actively look for peers on subnets. Unless by chance we find validator nodes, I imagine a standard beacon node will therefore always be in emergency mode (or at least very likely). 

Even when we have a validator attached, we only subscribe to one subnet per validator and again won't actively try and find peers for all subnets, only the ones we need. 

It seems to be the emergency mode would be more or less the standard state and having enough peers on every subnet would be an exceptional case (despite how the current testnets perform).

Reply:
We discussed emergency mode already in Mattermost and decided to remove it, but just for the sake of the argument here: We only consider topics (subnets) that we are subscribed to to decide if we are in emergency mode, therefore this should also work for non-validators in my opinion.